src/test/java/org/assertj/core/api/ComparatorFactory_floatComparatorWithPrecision.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Comparator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class ComparatorFactory_floatComparatorWithPrecision {
+
+  private final ComparatorFactory INSTANCE = ComparatorFactory.INSTANCE;
+
+  @Test
+  void should_pass_for_expected_equal_to_actual_in_certain_range1() {
+    Comparator<Float> comparator0_01f = INSTANCE.floatComparatorWithPrecision(0.01f);
+
+    assertThat(0).isEqualTo(comparator0_01f.compare(0.111f, 0.121f));
+    assertThat(0).isEqualTo(comparator0_01f.compare(0.121f, 0.111f));
+    assertThat(1).isEqualTo(comparator0_01f.compare(0.1211f, 0.111f));
+    assertThat(-1).isEqualTo(comparator0_01f.compare(0.111f, 0.1211f));
```

Comment:
you are inverting actual and expected here, `comparator0_01f.compare(0.111f, 0.1211f)` is the object under test (actual) whereas `-1` is the expected value., this is why writing test with GIVEN WHEN THEN helps clarifying what is being tested and what is expected.

Replace should_pass_for_expected_equal_to_actual_in_certain_range1 and should_pass_for_expected_equal_to_actual_in_certain_range2 by a parameterized test as I already suggested

this should look like (with an additional parameter for precision)
```
  @ParameterizedTest
  @CsvSource({ "1.0f, 1.0f", "0.0f, 0.0f" })
  void should_pass_using_primitive_comparison(Float actual, Float expected) {
    then(actual).isEqualTo(expected);
  }
```

Reply:
I have fixed this problem.